lib/elixir/test/elixir/kernel/overridable_test.exs

```diff
@@ -92,6 +92,29 @@ defmodule Kernel.Overridable do
   end
 end
 
+defmodule Kernel.ExampleBehaviour do
+  @callback required_callback :: any
+  @optional_callbacks optional_callback: 0
```

Comment:
An optional_callback still needs to be defined as a callback. I am surprised this does not raise any error.